frontend/apps/reader/modules/readerpaging.lua

```diff
@@ -903,6 +922,83 @@ function ReaderPaging:onGotoPageRel(diff)
             return true
         end
 
+    elseif self.zoom_mode:find("pan") then
+        -- pan zoom mode
+        logger.dbg("ZOOM: Pan mode")
+
+        local panning_update = {x = nil, y = nil}
+        local page_area, old_va = self.page_area, self.visible_area
+        local x, y, w, h, left = "x", "y", "w", "h", "left"
+        local h_progress = 1 - self.zoom_pan_h_overlap / 100
+        local v_progress = 1 - self.zoom_pan_v_overlap / 100
+        local page_diff = diff
+
+        if self.zoom_pan_direction_vertical then  -- invert axes
+            y, x, h, w, left = x, y, w, h, "top"
+            h_progress = 1 - self.zoom_pan_v_overlap / 100
+            v_progress = 1 - self.zoom_pan_h_overlap / 100
+        end
+
+        if self.zoom_pan_right_to_left then diff = -diff end
+
+        x_pan_off = Math.roundAwayFromZero(self.visible_area[w] * h_progress * diff)
+        y_pan_off = Math.roundAwayFromZero(self.visible_area[h] * v_progress * diff)
```

Comment:
Regarding my comment about problems with bottom -> top. By setting `diff = -diff` doesn't that "break" the y_pan_off?

Reply:
Should be fixed now; may you check it please ?